docs/products/clickhouse/concepts/indexing.rst

```diff
@@ -0,0 +1,21 @@
+Indexing and data processing
+=============================
+
+In this article we look at how ClickHouse works with data and how this differs from other database management systems.
+In particular we'll touch on such topics as **data indexing** and **a vector computation engine**.
+
+Understanding ClickHouse approach to data processing will help you learn why ClickHouse is perfect solution for some use cases and an awful choice for others.
+
+ClickHouse is :doc:`a columnar database <columnar-databases>`, this means that ClickHouse can read columns selectively retrieving only necessary information and omitting columns that are not needed for the request. The smaller the number of the columns you read, the faster and more efficient is the performance of the request. If you have to read many or all columns, then using columnar database becomes highly ineffective and we recommend reconsidering this solution.
+
+ClickHouse is designed to process massive chunks of information for a single request. To make this possible it shifts the focus from reading individual lines into scanning massive blocks of data for a request. This happens by using what is called **a vector computation engine**, which reads data in blocks that normally consist of thousands of rows, selected for a small set of columns.
+
+To find a specific block, ClickHouse uses **a primary index**. However, unlike in other systems the primary index in ClickHouse has a couple of distinctive characteristics.
```

Comment:
I would add a subtitle like `ClickHouse primary index` and put the ordering sentence before, since the primary index mechanism described before works only if the column is ordered beforehand

Reply:
I am lost as well, so I'm resolving the comment and hoping for the best!